server/src/service.rs

```diff
@@ -102,24 +103,31 @@ pub struct ServiceState {
     project_cache: Addr<ProjectCache>,
     upstream_relay: Addr<UpstreamRelay>,
     event_manager: Addr<EventManager>,
+    outcome_producer: Addr<OutcomeProducer>,
 }
 
 impl ServiceState {
     /// Starts all services and returns addresses to all of them.
     pub fn start(config: Config) -> Result<Self, ServerError> {
         let config = Arc::new(config);
         let upstream_relay = UpstreamRelay::new(config.clone()).start();
-
-        let event_manager = EventManager::create(config.clone(), upstream_relay.clone())
-            .context(ServerErrorKind::ConfigError)?
-            .start();
+        let outcome_producer = OutcomeProducer::create(config.clone())?.start();
```

Comment:
This is inconsistent with how StoreForwarder works, because `OutcomeProducer` resolves to a noop internally while StoreForwarder is stored as `Option<>` and only instantiated if processing is enabled. I would like to do the same for OutcomeProducer because I hope to have a single place in `EventManager::create` that does the check for `Is_processing_enabled` only once. The other reason is that I already applied StoreForwarder's idiom to quotas as well.

Having the callsite of `track_outcome` check for the `None`-ness of your `Option<Addr<OutcomeProducer>>` will also help avoid creating unnecessary structs and strings when processing is disabled. I'd also argue that `if let Some(producer) = outcome_producer { }` everywhere is not as much code as having the noop implementation

Reply:
Yes you are right, it is inconsistent with the way StoreForwarder was implemented. We need to decide if it is worthwhile changing the implementation of OutcomeProducer for this.

Reply:
Discussed offline. In this case we agree that we use a different pattern. In fact, `OutcomeProducer` will start to emit a debug log for every outcome now, and may potentially emit metrics in the future. 

Thanks for raising this @untitaker!